### ISSUE TITLE
Mapping pbmm2 Presets to Minimap2 — Clarification on Homopolymer-Compressed K-Mers

### ISSUE THREAD

**Issue by CharlesARoy** [lh3/minimap2#769]
Hi @lh3 @armintoepfer,

As described in [this issue](https://github.com/PacificBiosciences/pbbioconda/issues/415) I am attempting to align PacBio CCS (HiFi) data _and include secondary alignments in the output_. pbmm2 is much more convenient for handling PacBio data but it doesn't have the option to output secondary alignments. Given this, here's what I'm wondering:

1. The [pbmm2 documentation](https://github.com/PacificBiosciences/pbmm2) mentions different presets for aligning subread data vs CCS/HiFi data but the presets listed in the [Minimap2 documentation](https://lh3.github.io/minimap2/minimap2.html) don't appear to distinguish between these, is that correct?

2. If the pbmm2 presets for CCS data are better than the Minimap2 v 2.20 presets, are the following settings correct replacements in Minimap2 to achieve the same alignment parameters as pbmm2? (The discrepancies appear to stem from the fact that pbmm2 uses Minimap2 v 2.15 whereas I am using Minimap2 v 2.20). **_Most importantly_**, am I correct in my understanding that -u which is used to disable homopolymer-compressed k-mers in pbmm2 is now the default in Minimap2 (i.e. using -u in pbmm2 is the same as **_not_** using -H in Minimap2)?

| Preset Parameter in pbmm2 | Meaning in pbmm2                                          | Meaning in Minimap2                             | Likely Replacement Parameter in Minimap2                                                                       |
|---------------------------|-----------------------------------------------------------|-------------------------------------------------|----------------------------------------------------------------------------------------------------------------|
| -k 19                     | k-mer size                                                | Same                                            | -k 19                                                                                                          |
| -w 10                     | Minimizer window size                                     | Same                                            | -w 10                                                                                                          |
| -u                        | Disable homopolymer-compressed k-mer                      | How to find canonical splicing sites            | Remove parameter (Homopolymer-compressed k-mers appear to be disabled by default)                              |
| -o 5                      | Gap open penalty 1                                        | Output alignments to FILE                       | Parameter replaced by -O                                                                                       |
| -O 56                     | Gap open penalty 2                                        | Gap open penalty 1 and 2                        | -O 5,56                                                                                                        |
| -e 4                      | Gap extension penalty 1                                   | None                                            | Parameter replaced by -E                                                                                       |
| -E 1                      | Gap extension penalty 2                                   | Gap extension penalty 1 and 2                   | -E 4,1                                                                                                         |
| -A 2                      | Matching score                                            | Same                                            | -A 2                                                                                                           |
| -B 5                      | Mismatch penalty                                          | Same                                            | -B 5                                                                                                           |
| -z 400                    | Z-drop score                                              | Z-drop score and inversion Z-drop score         | -z 400,50                                                                                                      |
| -Z 50                     | Z-drop inversion score                                    | None                                            | Parameter replaced by -z                                                                                       |
| -r 2000                   | Bandwidth used in chaining and DP-based alignment         | Same                                            | -r 2000                                                                                                        |
| -L 0.5                    | Long join flank ratio                                     | Write CIGAR with >65535 operators at the CG tag | --lj-min-ratio 0.5 (while -L appears to have a new meaning, --lj-min-ratio appears to be the same in Minimap2) |
| -g 5000                   | Stop chain enlongation if there are no minimizers in N bp | Same                                            | -g 5000                                                                                                        |

<br>

In summary, if I am correct about everything, the pbmm2 command:
`pbmm2 -k 19 -w 10 -u -o 5 -O 56 -e 4 -E 1 -A 2 -B 5 -z 400 -Z 50 -r 2000 -L 0.5 -g 5000`

would map to this command in Minimap2 v2.20:
`minimap2 -k 19 -w 10 -O 5,56 -E 4,1 -A 2 -B 5 -z 400,50 -r 2000 --lj-min-ratio 0.5 -g 5000`

**Comment by lh3** [comment]
1. In minimap2, `map-pb` is for CLR and `map-hifi` for HiFi reads. The manpage you saw was for v2.18. I have just updated it to v2.20. The explanation is clearer there. Also see README. It gives the recommended setting for different types of data.

2. I don't use pbmm2, so don't know what is the role of `-u`. You have to ask pbmm2 developers. Minimap2 doesn't apply homopolymer compression by default. Also note that pbmm2 lags behind minimap2. `--lj-min-ratio` has no effect with minimap2 v2.20. I recommend v2.20 over pbmm2 due to recent improvements.
